types/coin.go

```diff
@@ -226,6 +243,37 @@ func (coins Coins) safePlus(coinsB Coins) Coins {
 	}
 }
 
+// SubCoinByDenom subtracts a single coin from another coin in a set of coins by
+// denomination. If the coins are empty, the coin is added to the empty set. If
+// the resulting coin is zero, it is removed.
+func (coins Coins) SubCoinByDenom(other Coin) Coins {
+	if len(coins) == 0 {
+		return Coins{other}
+	}
+
+	for i, coin := range coins {
+		if coin.Denom == other.Denom {
+			otherNeg := Coin{other.Denom, other.Amount.Neg()}
+
+			coins[i] = coin.Plus(otherNeg)
+			if coins[i].IsZero() {
+				// remove resulting zero coin
+				coins = append(coins[:i], coins[i+1:]...)
+			} else if !coins[i].IsNotNegative() {
```

Comment:
hmmm not related for this scope but that function should be `IsNegative()`

Reply:
Dude I know, but we're moving to unsigned integers anyway. If we do it, I agree that it should be in another PR.